# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/VideoBridgeParent.h
# Commit: 5a60cd493115
# Full Hash: 5a60cd493115214a5a3cf1cd0037e25ae6645e2c
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-06-28 16:12:48
# Regressor Bug: 1561178
# File Overlap Count: 1
# Description:
#   Bug 1561178 - Use Endpoints for VideoBridge construction, and support the possibility that there are multiple at the same time. r=jya
#   
#   In the future we're going to want VideoBridge connections from the RDD process into both the parent process and the GPU process.
#   This does the preparation work for unifying the way we create VideoBridges (using Endpoints, required for cross-process connections),
#   and detects which one to use based on where the video will be composited.
# ==============================================================================

diff -r c9f9bba02acf -r 5a60cd493115 gfx/layers/ipc/VideoBridgeParent.h
--- a/gfx/layers/ipc/VideoBridgeParent.h	Fri Jun 28 06:32:16 2019 +0000
+++ b/gfx/layers/ipc/VideoBridgeParent.h	Fri Jun 28 07:00:41 2019 +0000
@@ -19,7 +19,6 @@
                                 public HostIPCAllocator,
                                 public ShmemAllocator {
  public:
-  VideoBridgeParent();
   ~VideoBridgeParent();
 
   static VideoBridgeParent* GetSingleton();
@@ -55,7 +54,11 @@
 
   void DeallocShmem(ipc::Shmem& aShmem) override;
 
+  static void Open(Endpoint<PVideoBridgeParent>&& aEndpoint);
+
  private:
+  VideoBridgeParent();
+
   void ActorDealloc() override;
 
   // This keeps us alive until ActorDestroy(), at which point we do a
